clarity/src/vm/costs/mod.rs

```diff
@@ -811,9 +811,10 @@ impl TrackerData {
 
         self.contract_call_circuits = contract_call_circuits;
 
-        let mut cost_contracts = HashMap::new();
-        let mut m = HashMap::new();
-        for f in ClarityCostFunction::ALL.iter() {
+        let iter = ClarityCostFunction::ALL.iter();
```

Comment:
Probably better to put the `len()` in a variable instead of calling it twice - no guarantees the backing type has an efficient `len()` implementation.

Reply:
Good point. This is also why Clippy complains when you use `var.len() == 0` instead of `var.is_empty()`